GUIDELINES.md

```diff
@@ -0,0 +1,118 @@
+
+- Trait names can leverage their namespace and package name in order to
+  reduce ambiguity, vs attempting the futile quest to be globally unique.
+
+- Trait property names should describe the nature of the value as
+  precisely as possible. They shouldn't require the user to consult the
+  documentation. For example:
+  - Good: `framesPerSecond`, `metersPerLinearUnit`
+  - Bad: `frameRate`, `units`
+
+- "Family" traits should be defined to allow high-level partitioning of
+  entity space. For example the `Image` and `Spatial` traits exist
+  solely to allow users to broadly filter 2D and 3D entities without
+  needing multiple top-level trait sets. Care should be taken when
+  defining properties for these traits, to ensure they are genuinely
+  applicable to all entities with the quality in question. Eg. `Spatial`
+  does not include a bounding box as there are many unbounded or
+  zero-volume concepts in the 3D world, this should be expressed through
+  a peer trait in the `threeDimensional` namespace that can be composed
+  as needed.
+
+- The absence of a trait _does not_ imply its inverse, and so companion
+  traits may be needed. For example the lack of a `FrameRanged` trait
+  does not imply the entity is not animated. The absence of a trait that
+  defines any quality from a set means _the entity does not specify
+  whether it exhibits that quality or not_. This allows trait sets to be
+  used as filter predicates:
+  - `{ Entity, Image }`: All images (static, animated or any other).
+  - `{ Entity, Image, Static }`: Only static images without animation.
+  - `{ Entity, Image, FrameRanged }`: Only animated images that cover a
+    discrete series of frames.
+
+## Specifications
+
+Specifications define well-known sets of traits that map to first-class
+concepts within our workflows. They are used to define well-known
+entities ("assets") and relationships between them.
+
+- Specifications should contain the minimum number of traits required to
+  define a particular concrete workflow entity. Trait sets are
+  [additive](https://openassetio.github.io/OpenAssetIO/entities_traits_and_specifications.html#Specifications),
+  and so each trait stipulates that the entity _must_ possess that
+  quality, _and_ the qualities of all the other traits in the set. There
+  is no way to define a specification that states "one trait or the
+  other".
+
+- Specification names can leverage their namespace and package name in order to
+  reduce ambiguity, vs attempting the futile quest to be globally unique.
+
+- "Family" specifications should be defined to allow high-level
+  partitioning of entity space. For example the `Image` and `Spatial`
+  specifications exist solely to allow users to broadly filter 2D and 3D
+  entities without needing multiple top-level specifications. Care should be
+  taken to ensure that their trait sets are not overly constrained to
+  preclude future evolution. For example, the aforementioned
```

Comment:
[consider][minor] Took me a sec to know what you meant by "aforementioned" (at first I thought it meant all Specifications). Perhaps worth explicitly saying `Image` and `Spatial`.